libvast/src/format/reader_factory.cpp

```diff
@@ -63,9 +60,30 @@ void factory_traits<format::reader>::initialize() {
   fac::add("zeek", make_reader<zeek::reader>);
   fac::add("zeek-json",
            make_reader<format::json::reader<format::json::zeek_selector>>);
-#if VAST_ENABLE_PCAP
-  fac::add("pcap", make_reader<pcap::reader>);
-#endif
+  for (const auto& plugin : plugins::get()) {
+    if (const auto* reader = plugin.as<reader_plugin>()) {
+      fac::add(
+        reader->reader_format(),
+        [name = std::string{plugin->name()}](const caf::settings& options)
+          -> caf::expected<std::unique_ptr<format::reader>> {
+          for (const auto& plugin : plugins::get()) {
+            if (plugin->name() != name)
+              continue;
+            const auto* reader = plugin.as<reader_plugin>();
+            VAST_ASSERT(reader);
+            auto in = detail::make_input_stream(options);
+            if (!in)
+              return in.error();
```

Comment:
Move these into every reader plugins' `make_reader` (and do the same for writer plugins and `make_writer`).